ksqldb-functional-tests/src/test/resources/query-validation-tests/group-by.json

```diff
@@ -2272,6 +2313,277 @@
+        "CREATE STREAM TEST (ID INT KEY, A INT, B INT) WITH (kafka_topic='test_topic', format='JSON');",
+        "CREATE TABLE OUTPUT WITH (key_format='AVRO') AS SELECT STRUCT(a:=A, b:=B) AS ROWKEY, COUNT() AS COUNT FROM TEST GROUP BY STRUCT(a:=A, b:=B);"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": {"A": 1, "B": 1}},
+        {"topic": "test_topic", "key": 1, "value": {"A": 2, "B": 1}},
+        {"topic": "test_topic", "key": 2, "value": {"A": 1, "B": 1}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": {"A": 1, "B": 1}, "value": {"COUNT": 1}},
+        {"topic": "OUTPUT", "key": {"A": 2, "B": 1}, "value": {"COUNT": 1}},
+        {"topic": "OUTPUT", "key": {"A": 1, "B": 1}, "value": {"COUNT": 2}}
+      ],
+      "post": {
+        "topics": {
+          "topics": [
+            {
+              "name": "OUTPUT",
+              "keyFormat" : { "format" : "AVRO", "features": ["UNWRAP_SINGLES"] },
+              "keySchema": {
+                "type": "record",
+                "name": "KsqlDataSourceSchema",
+                "namespace": "io.confluent.ksql.avro_schemas",
+                "fields": [
+                  { "name": "A", "type": [ "null", "int" ], "default": null },
+                  { "name": "B", "type": [ "null", "int" ], "default": null }
+                ]
+              },
+              "valueFormat" : { "format" : "JSON" }
+            }
+          ]
+        }
+      }
+    },
+    {
+      "properties": {
+        "ksql.key.format.enabled": "true"
+      },
+      "name": "JSON group by struct convert to incompatible key format",
+      "statements": [
+        "CREATE STREAM TEST (ID INT KEY, A INT, B INT) WITH (kafka_topic='test_topic', format='JSON');",
+        "CREATE TABLE OUTPUT WITH (key_format='DELIMITED') AS SELECT STRUCT(a:=A, b:=B) AS ROWKEY, COUNT() AS COUNT FROM TEST GROUP BY STRUCT(a:=A, b:=B);"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The 'DELIMITED' format does not support type 'STRUCT', column: `ROWKEY`"
+      }
+    },
+    {
+      "properties": {
+        "ksql.key.format.enabled": "true"
+      },
+      "name": "Struct key used in aggregate expression",
+      "statements": [
+        "CREATE STREAM TEST (ID STRUCT<F1 INT> KEY, VAL INT) WITH (kafka_topic='test_topic', format='JSON');",
+        "CREATE TABLE OUTPUT AS SELECT ID, SUM(ID->F1) AS sum FROM TEST GROUP BY ID;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": {"F1": 1}, "value": {"VAL": 1}},
+        {"topic": "test_topic", "key": {"F1": 2}, "value": {"VAL": 2}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": {"F1": 1}, "value": {"SUM": 1}},
+        {"topic": "OUTPUT", "key": {"F1": 2}, "value": {"SUM": 3}}
+      ]
+    },
+    {
+      "properties": {
+        "ksql.key.format.enabled": "true"
+      },
+      "name": "Struct key used in aggregate expression",
```

Comment:
This test looks identical to the one above it?

Reply:
eh, I forgot to remove the one above it (which actually fails - the data is different)